testing/interceptor_suite.go

```diff
@@ -143,3 +161,76 @@ func (s *InterceptorTestSuite) TearDownSuite() {
 		s.clientConn.Close()
 	}
 }
+
+type keycert struct {
+	Key, Cert string
+}
+
+func generateTestCerts() error {
+	oldDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	newDir := getTestingCertsPath()
+	if newDir == "" {
+		return errors.New("error generating test certificates - unable to get testing certs path")
+	}
+	if err := os.Chdir(newDir); err != nil {
+		return errors.New("error generating test certificates - unable to chdir to " + newDir)
+	}
+
+	defer os.Chdir(oldDir)
+	if fileExists("cert.pem") && fileExists("key.pem") {
+		return nil
+	}
+	goroot := runtime.GOROOT()
+	cmd := exec.Command(filepath.Join(goroot, "bin/go"),
+		"run",
+		filepath.Join(goroot, "src/crypto/tls/generate_cert.go"),
+		"--ca",
+		"--rsa-bits", "2048",
+		"--host", "localhost,example.com",
+		"--start-date", "Jan 1 00:00:00 2020",
+		"--duration=1000000h")
+	if _, err := cmd.CombinedOutput(); err != nil {
+		return err
+	}
+	keyBytes, err := ioutil.ReadFile("key.pem")
+	if err != nil {
+		return err
+	}
+	certBytes, err := ioutil.ReadFile("cert.pem")
+	if err != nil {
+		return err
+	}
+	data := keycert{
+		Key:  string(keyBytes),
+		Cert: string(certBytes),
+	}
+	// localhost.go template
+	const tmpl = `package certs
+
+var LocalhostKey = []byte(` + "`" + `{{.Key}}` + "`" + `)
+
+var LocalhostCert = []byte(` + "`" + `{{.Cert}}` + "`" + `)
+`
+
+	t := template.Must(template.New("keycert-go").Parse(tmpl))
+	f, err := os.OpenFile("localhost.go", os.O_RDWR|os.O_TRUNC|os.O_CREATE, 0644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	if err := t.Execute(f, data); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
I think we could replace this whole function with one that generates a certificate and key and then returns the in-memory representation. Now that we're doing it in the application we shouldn't need to use the filesystem at all. See something like https://github.com/johanbrandhorst/certify/blob/master/issuers/vault/vault_suite_test.go#L255.

Reply:
We can then use https://pkg.go.dev/google.golang.org/grpc@v1.31.1/credentials?tab=doc#NewClientTLSFromCert and https://pkg.go.dev/google.golang.org/grpc@v1.31.1/credentials?tab=doc#NewServerTLSFromCert.

Reply:
removed the filesystem usage - had to use this snippet instead of `NewClientTLSFromCert` - similar to https://github.com/grpc/grpc-go/blob/v1.31.1/credentials/tls.go#L191
```
		cp := x509.NewCertPool()
		if !cp.AppendCertsFromPEM(testcert.CertPEM()) {
			s.T().Fatal("failed to append certificate")
		}
		creds := credentials.NewTLS(&tls.Config{ServerName: "localhost", RootCAs: cp})
```